packages/app-staking/src/Actions/Accounts.tsx

```diff
@@ -31,13 +33,40 @@ const Wrapper = styled(CardGrid) `
 
 class Accounts extends React.PureComponent<Props,State> {
   state: State = {
-    isNewStakeOpen: false
+    isNewStakeOpen: false,
+    myStashes: []
   };
 
+  async componentWillReceiveProps ({ allAccounts, api }: Props, { isNewStakeOpen }: State) {
```

Comment:
Yes, I hate it. The issue here is that -

- we can certainly simplify
- this will be done each time it is called (put a console.log in here, it will probably be ugly with the number of times called)

So short-term -

- you could store the allAccounts keys in state as well, checking first if it has changed before calling
- def don't make this async, rather in the `.then` set the state
- at the very least, needs a multi call instead of single calls

The correct approach probably would be -

- add something like a `pickProps(props: Props)` function alongside the `transform` in api
- use this to transform the keys
- make a multi-call